wicket-core/src/main/java/org/apache/wicket/protocol/http/CsrfPreventionRequestCycleListener.java

```diff
@@ -110,11 +110,15 @@
  * <li>{@link #onSuppressed(HttpServletRequest, String, IRequestablePage)} when an origin was in
  * conflict and the request should be suppressed</li>
  * </ul>
+ *
+ * @see FetchMetadataRequestCycleListener
+ * @deprecated
  */
-public class CsrfPreventionRequestCycleListener implements IRequestCycleListener
+@Deprecated(since = "XXX")
```

Comment:
@svenmeier any thoughts about what version should go here?

Reply:
"9.1.0"
But IMO it needs some work before being deprecated. The logic should be extracted in a new class implementing ResourceIsolationPolicy that this listener should extend from. This way we could remove this class for Wicket 10 (as we do with all deprecated classes in a major version release).

Reply:
This solution looks much better now. I've created a new OriginBasedResourceIsolationPolicy, so Wicket will be able to defend against CSRF attacks on modern and legacy browsers. I've set 9.1.0 as you said but would be happy to iterate again if you think the Deprecated annotation shouldn't be here yet. 

Thanks for your review!